src/rsa.rs

```diff
@@ -475,6 +486,51 @@ extern {
                                  rng: *mut rand::RAND) -> c::int;
 }
 
+#[cfg(feature = "rsa_signing")]
+#[allow(non_snake_case)]
+#[doc(hidden)]
+#[no_mangle]
+pub unsafe extern fn BN_rand_range(dest: *mut BIGNUM, range: *const BIGNUM,
+                                   rng: *mut rand::RAND) -> c::int {
```

Comment:
> (B) We leave the definition of BN_rand_range in C, and call into a Rust function internally. I'm neutral on this; it seems the advantage is we have a smaller diff, and the disadvantage is we can't delete random.c yet. If it helps us make (A) work, though, I'm for it.

I think this is a good first step because it lets us review the change in parts.

Also, you can imagine that after this PR, we'll want to replace `BN_BLINDING` stuff with Rust code and then replace `GFp_private_key_transform` with Rust code. This is actually a step towards all of that. And, I think the question is, in between now and then, do we want to have `#[repr(C)] struct BIGNUM {...}` in the Rust code? I don't think we do. I think we should say "old code uses BIGNUM as it needs to, new code doesn't use BIGNUM at all" and in particular, leave all the stuff that relies on a `BIGNUM` internals in C. Then, later, we'll get rid of `BIGNUM` completely.